src/lib/api/types.d.ts

```diff
@@ -33,11 +33,18 @@ export type ViewerMode = ReadMode | WriteMode;
 
 export type Zoom = number | Sizes | "width" | "height";
 
-export interface APIError {
-  error: {
-    status: number;
-    message: string;
-  };
+export interface APIError<E> {
+  status: number;
+  message: string;
+  errors?: E;
+}
+
+/**
+ * Wrap an API response so we can pass errors along
+ */
+export interface APIResponse<T, E> {
```

Comment:
By providing a default here, we don't need to specify the error type if we don't know it:

```suggestion
export interface APIResponse<T, E = unknown> {
```
`unknown` is a stricter type than `any`